test/unit-test/stun_serializer/stun_serializer_utest.c

```diff
@@ -149,3 +149,318 @@ void test_StunSerializer_Init_BadParams( void )
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate StunSerializer_AddAttributeErrorCode when pCtx->pStart is null.
+ */
+void test_StunSerializer_AddAttributeErrorCode_pStartNull( void )
+{
+    StunContext_t ctx = { 0 };
+    StunResult_t result;
+    StunHeader_t header = { 0 };
+    uint8_t buffer[ MAX_BUFFER_LENGTH ] = { 0 };
+    uint8_t transactionId[ STUN_HEADER_TRANSACTION_ID_LENGTH ] = { 0x12, 0x34, 0x56,
+                                                                   0x78, 0x9A, 0xBC,
+                                                                   0xDE, 0xF0, 0xAB,
+                                                                   0xCD, 0xEF, 0xA5 };
+    const uint8_t * errorPhrase = ( const uint8_t * ) "Error Phrase";
+    uint16_t errorPhraseLength = sizeof(errorPhrase );
+    uint16_t errorCode = 0x0600;
+
+    header.messageType = STUN_MESSAGE_TYPE_BINDING_REQUEST;
+    header.pTransactionId = &( transactionId[ 0 ] );
+
+     result = StunSerializer_Init( &( ctx ),
+                                  &( buffer[0] ),
+                                  MAX_BUFFER_LENGTH,
+                                  &( header ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
+
+    ctx.pStart = NULL; /* Set pCtx->pStart to NULL */
+
+    result = StunSerializer_AddAttributeErrorCode( &(ctx),
+                                                   errorCode,
+                                                   &(errorPhrase[0]),
+                                                   errorPhraseLength );
+                                                   
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate StunSerializer_AddAttributeErrorCode in case of out of memory.
+ */
+void test_StunSerializer_AddAttributeErrorCode_OutOfMemory( void )
+{
+    StunContext_t ctx = { 0 };
+    StunResult_t result;
+    StunHeader_t header = { 0 };
+    uint8_t buffer[ 20 ] = { 0 };
+    uint8_t transactionId[ STUN_HEADER_TRANSACTION_ID_LENGTH ] = { 0x12, 0x34, 0x56,
+                                                                   0x78, 0x9A, 0xBC,
+                                                                   0xDE, 0xF0, 0xAB,
+                                                                   0xCD, 0xEF, 0xA5 };
+    const uint8_t * errorPhrase = ( const uint8_t * ) "Error Phrase";
+    uint16_t errorPhraseLength = strlen( ( const char * ) errorPhrase );
+    uint16_t errorCode = 0x0600;
+
+    header.messageType = STUN_MESSAGE_TYPE_BINDING_REQUEST;
+    header.pTransactionId = &( transactionId[ 0 ] );
+
+    result = StunSerializer_Init( &( ctx ),
+                                  &( buffer[0] ),
+                                  20,
```

Comment:
Is this calculated backwards using the `errorPhaseLength` value? For readability can we add a small comment here , how this leads to Out of Memory error code?

Reply:
Actually, this 20 is just the fixed-length header..on trying to add an error attribute the buffer will overflow.